tajo-core/src/main/java/org/apache/tajo/querymaster/Query.java

```diff
@@ -402,6 +403,39 @@ public QueryState getSynchronizedState() {
     }
   }
 
+  public boolean hasUnionPlan() throws Exception {
+    boolean result = false;
+
+    // During executing UNION statement or UNION ALL statement, the terminal block doesn't have join plan and has
+    // two more child blocks. In above case, Tajo should summarize result TableStats from all child stages.
+    ExecutionBlock terminalBlock = plan.getTerminalBlock();
+    terminalBlock.setPlan(terminalBlock.getPlan());
```

Comment:
The terminal block doesn't have any plan, so this line doesn't affect to anything. I wonder what is your intention.